### ISSUE TITLE
Cannot find merlin binary at "ocamlmerlin"

### ISSUE THREAD

**Issue by nickzuber** [ocaml/merlin#739]
I have uninstalled and reinstalled the `merlin` package using OPAM and can't seem to get it working. 

```
$ opam install merlin
[NOTE] Package merlin is already installed (current version is 2.5.4).
```

I get the warning message that `Cannot find merlin binary at "ocamlmerlin"` when I try to open an OCaml file in VSCode for example, so I think there may be something wrong with my merlin binary?

Where should this binary be? Doing some investigation, I find

```
$ opam config var share   
/Users/nick/.opam/system/share
$ ls -a /Users/nick/.opam/system/share/merlin
.   ..  vim
```
I don't see a binary in this directory — is this correct?

When I look at my PATH I _see_ a path to an `ocamlmerlin` binary and I can run `ocamlmerlin` just find from the command line

```
$ echo $PATH | grep merlin
/Users/nick/.opam/system/bin/ocamlmerlin (...)
$ ocamlmerlin -vnum
2.5.4
```

So I'm confused as to what could have gone wrong in my installation or why something like VSCode is unable to find the "ocamlmerlin" binary when it's in my PATH.

Not quite sure what's going wrong here, but it seems like a problem with my installation of merlin. Uninstalling and reinstalling both using OPAM and cloning the repository and building manually don't seem to solve this problem for me.

**Comment by dbuenzli** [comment]
[This discussion](https://discuss.ocaml.org/t/goto-definition-feature-not-working-in-hackwalys-ocaml-extension-on-vscode/402/15?u=dbuenzli) may help.

**Comment by nickzuber** [comment]
Ah I think this actually was a VSCode problem. In my settings I added an absolute path to the ocamlmerlin binary and it seemed to do the trick

```json
{
    "ocaml.merlinPath": "/Users/nick/.opam/system/bin/ocamlmerlin",
    "reason.path.ocamlmerlin": "/Users/nick/.opam/system/bin/ocamlmerlin"
}
```

**Comment by jayantbh** [comment]
^ that seemed to work for me.

However, I did not have the `"ocaml.merlinPath"` setting in my default settings, so I did not attempt to add this to user settings either. I'm not sure whether adding it or not adding it will break anything.

For anyone new to VS Code (like myself, this is how it may look like):
```
{
    //...
    "reason.path.ocamlmerlin": "/Users/<YOUR-HOME-DIRECTORY>/.opam/system/bin/ocamlmerlin"
}
```
![image](https://user-images.githubusercontent.com/7949047/38815641-0aedcd52-41b2-11e8-9627-8380e9af7300.png)
